### PR TITLE
Removed python naming conventions from C# samples 

### LINKED ISSUE #2155: C# samples should probably use C# and not Python naming conventions
I was reading a PR when suddenly I saw something a bit odd.

https://github.com/dotnet/machinelearning/blob/6b9f5893ea6830481a8779f81f2b723e9b5cd6cd/docs/samples/Microsoft.ML.Samples/Dynamic/KeyToValue_Term.cs#L42-L47

Note the underscores in the middle of a variable name. While in private we might elect to bend the rules a little bit, in our public API and also probably our samples, we probably ought to write them with [consideration of standard practices](https://docs.microsoft.com/en-us/dotnet/standard/design-guidelines/capitalization-conventions). This does not seem to be a justifiable or explainable diversion from the recommended practice, and I'm guessing it was added by mistake.

Each individual renaming is probably not huge, but might span several files, since the example listed above is just one of many examples I see in our samples project. So not a "hard" change, except possibly in scale?

### PR DESCRIPTION
**Pull request by daniel-loudon**
Fixes #2155 

Removed the python naming conventions in `docs/samples/Microsoft.ML.Samples`

### CODE REVIEW

File: docs/samples/Microsoft.ML.Samples/Dynamic/KeyToValueTerm.cs [hunk 1]
```diff
@@ -31,20 +31,20 @@ public static void KeyToValue_Term()
...
             string customizedColumnName = "CustomizedKeys";
-            var customized_pipeline = new WordTokenizingEstimator(ml, "Review")
+            var customizedPipeline = new WordTokenizingEstimator(ml, "Review")
                 .Append(new ValueToKeyMappingEstimator(ml, "Review", customizedColumnName, maxNumTerms: 10, sort: ValueToKeyMappingTransformer.SortOrder.Value));
 
             // The transformed data.
-            var transformedData_default = default_pipeline.Fit(trainData).Transform(trainData);
-            var transformedData_customized = customized_pipeline.Fit(trainData).Transform(trainData);
+            var transformedDataDefault = defaultPipeline.Fit(trainData).Transform(trainData);
+            var transformedDatacustomized = customizedPipeline.Fit(trainData).Transform(trainData);
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/KeyToValueTerm.cs]: `transformedDataCustomized `

File: docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs [hunk 2]
```diff
@@ -10,23 +10,23 @@ public class MatrixFactorizationExample
         // The variable _synthesizedMatrixFirstRowIndex indicates the integer that would be mapped to the first row index. If user data uses
         // 0-based indices for rows, _synthesizedMatrixFirstRowIndex can be set to 0. Similarly, for 1-based indices, _synthesizedMatrixFirstRowIndex
         // could be 1.
-        const int _synthesizedMatrixFirstColumnIndex = 1;
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs]: Let's keep these with the underscore (`_`), but mark them private.

File: docs/samples/Microsoft.ML.Samples/Dynamic/NgramExtraction.cs [hunk 3]
```diff
@@ -33,8 +33,8 @@ public static void NgramTransform()
             var twoCharsPipeline = charsPipeline.Append(ngramTwpPipeline);
 
             // The transformed data for pipelines.
-            var transformedData_onechars = oneCharsPipeline.Fit(trainData).Transform(trainData);
-            var transformedData_twochars = twoCharsPipeline.Fit(trainData).Transform(trainData);
+            var transformedDataOnechars = oneCharsPipeline.Fit(trainData).Transform(trainData);
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/NgramExtraction.cs]: `OneChars`

File: docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs (hunk 2 above)
Comment (daniel-loudon) [reply docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs]: Ok I didn't think any names should have an underscore regardless of access

Comment (rogancarr) [reply docs/samples/Microsoft.ML.Samples/Dynamic/MatrixFactorization.cs]: The convention we follow is that private variables start with underscores and public variables start with caps.

File: docs/samples/Microsoft.ML.Samples/Dynamic/NgramExtraction.cs [hunk 4]
```diff
@@ -29,12 +29,12 @@ public static void NgramTransform()
...
             var ngramOnePipeline = ml.Transforms.Text.ProduceNgrams("Chars", "CharsUnigrams", ngramLength:1);
             var ngramTwpPipeline = ml.Transforms.Text.ProduceNgrams("Chars", "CharsTwograms");
-            var oneCharsPipeline = charsPipeline.Append(ngramOnePipeline);
+            var oneCharPipeline = charsPipeline.Append(ngramOnePipeline);
             var twoCharsPipeline = charsPipeline.Append(ngramTwpPipeline);
 
             // The transformed data for pipelines.
-            var transformedData_onechars = oneCharsPipeline.Fit(trainData).Transform(trainData);
-            var transformedData_twochars = twoCharsPipeline.Fit(trainData).Transform(trainData);
+            var transformedDataOnechar = oneCharPipeline.Fit(trainData).Transform(trainData);
```
Comment (rogancarr) [inline docs/samples/Microsoft.ML.Samples/Dynamic/NgramExtraction.cs]: OneChar

**Comment by Ivanidzo4ka** [comment]
@daniel-loudon  Do you have plans to update this PR to latest master and resolve conflicts?

**Comment by Ivanidzo4ka** [comment]
@daniel-loudon  Thank you for updating this PR. 
It just looks like something went terribly wrong.
>Files changed 563

That doesn't look like proper merge with latest master.

**Comment by daniel-loudon** [comment]
> 
> 
> @daniel-loudon Thank you for updating this PR.
> It just looks like something went terribly wrong.
> 
> > Files changed 563
> 
> That doesn't look like proper merge with latest master.

Yeha

**Pull request closed without merging**